File: python/scrabble-score/scrabble_score.py
SCOREMAP = {
    "aeioulnrst": 1,
    "dg": 2,
    "bcmp": 3,
    "fhvwy": 4,
    "k": 5,
    "jx": 8,
    "qz": 10
}


def score(word):
    total = 0
    for c in word.lower():
        for ks, val in SCOREMAP.items():
            if c in ks:
                total += val
                break
    return total

File: ruby/scrabble-score/scrabble_score.rb
# easy to maintain -> easy to use with function
def init_score_map(score)
  score_map = {}
  score.each do |key, value|
    key.scan(/\w/).each { |char| score_map[char] = value }
  end
  score_map.freeze
end

# easy to maintain score configuration
SCORE = init_score_map(
  'AEIOULNRST' => 1,
  'DG' => 2,
  'BCMP' => 3,
  'FHVWY' => 4,
  'K' => 5,
  'JX' => 8,
  'QZ' => 10
)

class Scrabble
  attr_reader :chars

  def initialize(word)
    @chars = word.to_s.upcase.scan(/\w/)
  end

  def score
    chars.sum { |char| SCORE[char] }
  end

  def self.score(word)
    new(word).score
  end
end
